test/jdk/tools/launcher/UnicodeTest.java

```diff
@@ -163,7 +163,7 @@ static String generateSources() throws Exception {
         return commandLineClassName;
     }
 
-    private static final String defaultEncoding = Charset.defaultCharset().name();
+    private static final String defaultEncoding = System.getProperty("sun.jnu.encoding");
```

Comment:
It's a bit confusing to have "defaultEncoding" set to the this property. Maybe the test should be changed to run with -Dfile.encoding=COMPAT or else change the two usages of defaultEncoding in the test.

Reply:
Since the test is asserting UTF-8 path names, it is not affected by `file.encoding` being `COMPAT` or not. To make it less confusing, I renamed the field name to `filePathEncoding`.